saleor/graphql/product/mutations/products.py

```diff
@@ -1784,6 +1784,67 @@ def perform_mutation(cls, _root, info, product_id, images_ids):
         return ProductImageReorder(product=product, images=images)
 
 
+class ProductVariantReorder(BaseMutation):
+    product = graphene.Field(Product)
+    variants = graphene.List(ProductVariant)
+
+    class Arguments:
+        product_id = graphene.ID(
+            required=True,
+            description="Id of product that variants order will be altered.",
+        )
+        variants = graphene.List(
```

Comment:
We have two APIs for reordering from what I see now. The first approach with passing moves:
- `attributeReorderValues(attributeId: ID!, moves: [ReorderInput]!): AttributeReorderValues`
- `collectionReorderProducts(collectionId: ID!, moves: [MoveProductInput]!): CollectionReorderProducts`
- `productTypeReorderAttributes(moves: [ReorderInput]!, productTypeId: ID!, type: AttributeTypeEnum!): ProductTypeReorderAttributes`

And the second with passing a list of IDs in the new order:
- `productImageReorder(imagesIds: [ID]!, productId: ID!): ProductImageReorder`

Your new mutation uses the second approach, but I think we should everywhere use the first one with passing the list of particular reordering moves, is that right @patrys? The image reordering mutation should be in fact refactored as well (not in this PR ofc).

Reply:
Yes, we want all mutations to be safe when used in a concurrent manner (two clients making changes that should not conflict with each other).